carto/auth.py

```diff
@@ -26,7 +26,43 @@
     from urlparse import urlparse
 
 
-class APIKeyAuthClient(BaseAuthClient):
+class _UsernameGetter:
+    def get_user_name(self, base_url):
+        try:
+            url_info = urlparse(base_url)
+            # On-Prem:
+            #   /user/<username>
+            m = re.search('^/user/([^/]+)/.*$', url_info.path)
+            if m is None:
+                # Cloud personal account (org and standalone)
+                # <username>.carto.com
+                netloc = url_info.netloc
+                if netloc.startswith('www.'):
+                    netloc = netloc.split('www.')[1]
```

Comment:
Since `netloc` starts with `www.`, this is just removing the initial `www.` right?

In that case: `netloc = netloc[4:]` is equivalent, and faster.

Reply:
Same ^. If it's important, I'll do it, otherwise, I'll keep it untouched.